include/crocoddyl_msgs/inertial_parameters_subscriber.h

```diff
@@ -0,0 +1,114 @@
+// All rights reserved.
+///////////////////////////////////////////////////////////////////////////////
+
+#ifndef CROCODDYL_MSG_MULTIBODY_INERTIAL_PARAMETERS_SUBSCRIBER_H_
+#define CROCODDYL_MSG_MULTIBODY_INERTIAL_PARAMETERS_SUBSCRIBER_H_
+
+#include "crocoddyl_msgs/MultibodyInertialParameters.h"
+
+#include <mutex>
+#include <ros/node_handle.h>
+
+typedef crocoddyl_msgs::MultibodyInertialParameters MultibodyInertialParameters;
+typedef const crocoddyl_msgs::MultibodyInertialParameters::ConstPtr
+    &MultibodyInertialParametersSharedPtr;
+namespace crocoddyl_msgs {
+
+
+class MultibodyInertialParametersRosSubscriber {
+public:
+  EIGEN_MAKE_ALIGNED_OPERATOR_NEW
+
+  /**
+   * @brief Initialize the multi-body inertial parameters subscriber.
+   *
+   * @param[in] n_bodies  Number of bodies
+   * @param[in] topic     Topic name
+   */
+  MultibodyInertialParametersRosSubscriber(
+      const unsigned int n_bodies,
+      const std::string &topic = "/robot/multibody_inertial_parameters"):
+      n_bodies_(n_bodies), spinner_(2), has_new_msg_(false), is_processing_msg_(false), last_msg_time_(0.) {
+    ros::NodeHandle n;
+    sub_ = n.subscribe<MultibodyInertialParameters>(
+        topic, 1, &MultibodyInertialParametersRosSubscriber::callback, this,
+        ros::TransportHints().tcpNoDelay());
+    parameters.resize(10);
+    spinner_.start();
+    ROS_INFO_STREAM("Subscribing MultibodyInertialParameters messages on " << topic);
+  }
+
+  ~MultibodyInertialParametersRosSubscriber() = default;
+
+  /**
+   * @brief Get the latest inertial parameters
+   *
+   * @return  An std mape with the pair of body name, inertial parameters
+   */
+  
+  std::map<std::string, Eigen::VectorXd>
+  get_inertial_parameters() {
+    // start processing the message
+    is_processing_msg_ = true;
+    std::lock_guard<std::mutex> guard(mutex_);
+
+    for (std::size_t i = 0; i<n_bodies_; ++i){
+      const std::string &name =  msg_.parameters[i].body_name;
+      for (std::size_t j = 0; j<10; ++j){
+        parameters[j] = msg_.parameters[i].psi[j];
+      }
+      parameters_map[msg_.parameters[i].body_name] = parameters;
+    }
+    // finish processing the message
+    is_processing_msg_ = false;
+    has_new_msg_ = false;
+    return parameters_map;
+  }
+
+  /**
+   * @brief Indicate whether we have received a new message
+   */
+  bool has_new_msg() const { return has_new_msg_; }
+
+private:
+  ros::NodeHandle node_;
+  ros::AsyncSpinner spinner_;
+  ros::Subscriber sub_; //!< ROS subscriber
+  std::mutex mutex_;      ///< Mutex to prevent race condition on callback
+  MultibodyInertialParameters msg_;    //!< ROS message
+
+  bool has_new_msg_;       //!< Indcate when a new message has been received
+  bool is_processing_msg_; //!< Indicate when we are processing the message
+  double last_msg_time_;
+
+  unsigned int n_bodies_;
+  Eigen::VectorXd parameters;
+  std::map<std::string, Eigen::VectorXd> parameters_map;
```

Comment:
1. Rename `parameters` and `parameters_map`.
3. Use Vector10d instead of `Eigen::VectorXd`.
4. Follow the Google code style, i.e., name class variables as "XXX_` instead of `XXX`
```suggestion
  Vector10d p_tmp_;
  std::map<std::string, Vector10d> parameters_;
```

The `Inertia` structure should be implemen